src/filename_absolute.cxx

```diff
@@ -45,30 +45,44 @@ static inline int isdirsep(char c) {return c == '/';}
     \return 0 if no change, non zero otherwise
  */
 int fl_filename_absolute(char *to, int tolen, const char *from) {
-  return Fl::system_driver()->filename_absolute(to, tolen, from);
+  char cwd_buf[FL_PATH_MAX];    // Current directory
+  // get the current directory and return if we can't
+  if (!fl_getcwd(cwd_buf, sizeof(cwd_buf))) {
+    strlcpy(to, from, tolen);
+    return 0;
+  }
+  return Fl::system_driver()->filename_absolute(to, tolen, from, cwd_buf);
 }
 
+/** Makes a filename absolute to any other directory.
+ \param[out] to resulting absolute filename
+ \param[in]  tolen size of the relative filename buffer
+ \param[in]  from relative filename
+ \param[in]  base relative to this absolute path
+ \return 0 if no change, non zero otherwise
+ */
+int                                     // O - 0 if no change, 1 if changed
+fl_filename_absolute(char       *to,    // O - Relative filename
+                     int        tolen,  // I - Size of "to" buffer
+                     const char *from,  // I - Absolute filename
+                     const char *base) { // I - Find path relative to this path
+  return Fl::system_driver()->filename_absolute(to, tolen, from, base);
```

Comment:
Same here, how can a path be "absolute" to another path ...

More of such issues below. I'm **_absolute_**ly confused. Pun intended.

Reply:
After reading again... Maybe the issue is only confusing wording and the comment(s)?

Probably better: "Returns an absolute filename from another filename that is relative to a base directory."

Should we also mention that the "base directory" must be an absolute path (if given), otherwise it would be the current directory, or something like that.

Reply:
Yes, it's confusing. I had to reread this myself a fe times, and even though the code worked, the docs are wrong. Also, documenting parameters twice, once in the Doxy comment, and a gain two lines below in code, is irritating, so I deleted it. Hoping that then phrasing is better.